src/lib/steps/init.ts

```diff
@@ -33,73 +34,48 @@ interface UserPackage {
-
-      if (packageJson['ngPackage']) {
-        log.debug(`Found package file ${folderPathOrFilePath}`);
-
-        return {
-          packageJson: packageJson,
-          ngPackageJson: Object.assign({}, packageJson['ngPackage']),
-          basePath
-        };
-      } else if (await fileExists(path.resolve(basePath, 'ng-package.json'))) {
-
-        return {
-          packageJson: packageJson,
-          ngPackageJson: await readJson(path.resolve(basePath, 'ng-package.json')),
-          basePath
-        };
-      } else if ((await fileExists(path.resolve(basePath, 'ng-package.js')))) {
-
-        // TODO: enable this, see #278
-        if (4 == 4) {
-          throw new Error(`Reading 'ngPackage' from a .js file is not yet implemented.`);
-        }
-      }
+    const fullPath = path.isAbsolute(folderPathOrFilePath) ? folderPathOrFilePath : path.resolve(folderPathOrFilePath);
+    const basePath = pathStats.isDirectory() ? fullPath : path.dirname(fullPath);
 
-    } else if (pathStats.isFile()) {
-
-      const fileName = path.basename(folderPathOrFilePath);
-      const folderPath = path.dirname(folderPathOrFilePath);
-      if (fileName=== 'ng-package.json') {
-
-        return {
-          packageJson: await readJson(path.resolve(folderPath, 'package.json')),
-          ngPackageJson: await readJson(folderPathOrFilePath),
-          basePath: folderPath
-        };
-      } else if (fileName === 'package.json') {
-        const packageJson = await readJson(folderPathOrFilePath);
-        if (!packageJson['ngPackage']) {
-          throw new Error(`Cannot read a package from 'package.json' without 'ngPackage' property.`);
-        }
+    const packageJson = await readJson(path.join(basePath, 'package.json'));
+    const packageConfPathJson = path.join(basePath, 'ng-package.json');
+    const packageConfPathJs = path.join(basePath, 'ng-package.js');
 
-        return {
-          packageJson: packageJson,
-          ngPackageJson: Object.assign({}, packageJson['ngPackage']),
-          basePath: folderPath
-        };
-      } else if (path.extname(folderPathOrFilePath) === '.js') {
+    let packageConf: UserPackage | undefined = {
+      basePath,
+      packageJson,
+      ngPackageJson: {}
+    };
+
+    if (packageJson['ngPackage']) {
+      packageConf.ngPackageJson = { ...packageJson['ngPackage'] }
+    } else if (await fileExists(packageConfPathJson)) {
+      packageConf.ngPackageJson = await readJson(packageConfPathJson);
+    } else if ((await fileExists(packageConfPathJs))) {
+      packageConf.ngPackageJson = await import(packageConfPathJs);
+    } else {
+      packageConf = undefined;
+    }
 
-        // TODO: enable this, see #278
-        if (4 == 4) {
-          throw new Error(`Reading 'ngPackage' from a .js file is not yet implemented.`);
-        }
+    if (packageConf || pathStats.isDirectory()) {
+      // return even if it's undefined and use defaults when it's not a file
+      return packageConf;
```

Comment:
I don't understand this. `packageConf` is initailized, then set to `undefined` in the "else", then checked here for "truthy" ... but ok as long as it works!?! Oo :)

Reply:
Tried to keep it simly, maybe i can do it in reverse if you’d prefer. Ie start as undefined and if not undefined, i’ll assign properties to the object.

Reply:
Ill see what I can do tomorrow or maybe merge it and I’ll refactor it :)

Reply:
No, it's ok ... yes, maybe it would be better to initialize with `undefined` and then assign values if it finds something... matter of taste. I'll merge it for now, ok?